### DISCUSSION TITLE
interaction.update is not a function

### DISCUSSION THREAD

**Discussion by White2001Offl** [discordjs/discord.js#6618] [Q&A] [answered]
I was trying to update the interaction which i receive
but its giving me error

```
module.exports = {
    data: new SlashCommandBuilder()
        .setName('panel')
        .setDescription('Creates a new order panel'),
    async execute(interaction) {
        if(interaction.member.permissions.has(Permissions.FLAGS.ADMINISTRATOR)){
            const row = new MessageActionRow()
                .addComponents(
                    new MessageButton()
                        .setCustomId('order')
                        .setLabel('Create Order')
                        .setStyle('SUCCESS')
                        .setEmoji("🎟️")
                );
            const embed = new MessageEmbed()
                .setTitle("Devsly Order Creation")
                .setDescription(`Hi,\nWe're Excited to work with you.\nClick the below Button\nAnd I'll meet you in your DM`)
                .setTimestamp(Date.now())
                .setFooter("Devsly","https://cdn.discordapp.com/avatars/750585310873649282/79870cc941ed9ea06085795cf232f27a.png?size=64")
            await interaction.update({components: [row], embeds: [embed]});

        } else{
            await interaction.reply({content: "You're not a admin", ephemeral: true})
        }

    },
};
```

**TypeError: interaction.update is not a function**

Here is the about code. am I doing something wrong?

**Comment by thebigwolfy** [comment]
you use the function "interaction.editReply()" not use "interaction.update()", is not existe (i am french, sorry)

**Reply by White2001Offl** [reply]
https://discord.js.org/#/docs/main/stable/class/ButtonInteraction?scrollTo=update

But here, they say there is a update function.
Basically editReply is used to edit the response the bot sent.
But I dont want like that.
It has to send embed and button without wait let me send the image

![image](https://user-images.githubusercontent.com/68497957/132863353-bd28abf9-b61b-4401-9953-56a27c6fef77.png)

I dont want that to be visible.
If i send only message, it says interaction failed

**Comment by ImRodry** [comment, marked as answer]
You are working with a CommandInteraction, not a ButtonInteraction and these cannot be updated. If you want a ButtonInteraction you need to use a collector on the reply and then you can use .update

**Reply by White2001Offl** [reply]
oh makes sense.
so, how to achieve the thing which I explained in above reply.

**Reply by White2001Offl** [reply]
like a send function
which wont show that thing.

**Reply by White2001Offl** [reply]
or like, how we can pass an empty response that this interaction was success
and we send a normal message

**Reply by ImRodry** [reply]
I have no clue what you wanna do but you cannot pass empty responses. You're not even collecting the button interactions so you should start from there

**Reply by White2001Offl** [reply]
ok well, I found a way to work around.

```js
await interaction.channel.send({components: [row], embeds: [embed]});
await interaction.reply({content: "Panel Created Successfully", ephemeral: true})
``` 

by making this, it kinda achieves what I needed.
Thanks for the reply!
